### DISCUSSION TITLE
Sample request: Azure B2C?

### DISCUSSION THREAD

**Discussion by plettb** [dotnet/maui#3691] [General]
Is anyone taking sample requests?  I'd really love to see an example of a Maui app that integrates (requires) an Azure B2C login.

Thanks!

**Comment by VladislavAntonyuk** [comment]
This is about AD, but you can simply update Authority to use it with AD B2C : https://vladislavantonyuk.azurewebsites.net/articles/Azure-Active-Directory-authentication-in-.NET-MAUI

**Reply by plettb** [reply]
Thanks!  Unfortunately,
1. That example doesn't build on my machine ("Microsoft.Graphics.Canvas.winmd cannot be referenced. Referencing a Windows Metadata component directly when targeting .NET 5 or higher is not supported.").
2. I was hoping for an example that didn't require modification of each of the platforms.  I actually have a working example that I ported from Xamarin, but I thought maybe Maui would offer an even better way.

**Reply by VladislavAntonyuk** [reply]
1. make sure you build the sample on Windows. 
2. you still need some modification, because it is required by the package. the changes in the platform are only in manifest and MauiActivity. you had the same changes in Xamarin.

**Comment by bryanp-ramosoil** [comment]
@VladislavAntonyuk,

Let me first start by saying thank you for the sample. As someone new to Xamarin and Maui this is a huge help. If I may ask for additional direction. What about reusable tokens? How would you add the ability to validate a token when the app is re-launched so the user doesn't have to login each time.

Thanks!

**Reply by VladislavAntonyuk** [reply]
There is a method to retrieve token silently.
I will update sample app.

**Reply by bryanp-ramosoil** [reply]
Thank you so much!!

**Reply by VladislavAntonyuk** [reply]
I have updated the sample.
1. OnAppearing call `AcquireTokenSilent` (https://github.com/VladislavAntonyuk/MauiSamples/blob/main/Auth/MauiAuth/AuthPage.cs#L44)
2. Depending on the result you will see Login button (on click call `SignInInteractively`) or Logout button https://github.com/VladislavAntonyuk/MauiSamples/blob/main/Auth/MauiAuth/AuthPage.cs#L70

**Reply by bryanp-ramosoil** [reply]
Thanks for that quick response. Now it's on to adding API calls using those credentials. Thanks again!

Cheers

**Reply by isaacrlevin** [reply]
@VladislavAntonyuk have you tested with macOS yet? I took a look at the issue mentioned in your blog post and it isn't clear if MSAL supports macOS yet

**Reply by VladislavAntonyuk** [reply]
It worked until .NET MAUI preview 8.

It will definitely work with MacCatalyst when that issue is fixed

**Comment by enterprisemk** [comment]
Hola, 

what's the deal with the RegisterServices static method inside the DependencyExtensions static class?

Where is it called from?

Thanks for the example, nice work.
